consensus/src/dag/dag_driver.rs

```diff
@@ -151,12 +155,19 @@ impl DagDriver {
         let signature_builder =
             SignatureBuilder::new(node.metadata().clone(), self.epoch_state.clone());
         let cert_ack_set = CertificateAckState::new(self.epoch_state.verifier.len());
+        let latest_ledger_info = self
```

Comment:
carrying a db here is a bit ugly, I was thinking about caching one in the notifier adapter and rely on callback to update it, wdyt

Reply:
i think we anyhow need the StorageAdapter here right? we can just use that instead. we can cache within the storage adapter if necessary?

Reply:
sounds better, just expose a function on the DAGStorage?